cirq-google/cirq_google/serialization/circuit_serializer.py

```diff
@@ -670,7 +696,31 @@ def _deserialize_gate_op(
                 raise ValueError(f"dimensions {dimensions} for ResetChannel must be an integer!")
             op = cirq.ResetChannel(dimension=dimensions)(*qubits)
         elif which_gate_type == 'internalgate':
-            op = arg_func_langs.internal_gate_from_proto(operation_proto.internalgate)(*qubits)
+            parsed_as_stimcirq = False
+            msg = operation_proto.internalgate
+            if msg.module == _STIMCIRQ_MODULE:
+                # special handling for stimcirq
+                try:
+                    import stimcirq
```

Comment:
If stimcirq is not installed this would search sys.path for every deserialized gate.

Consider moving this to a `@functools.cache` decorated function which returns `stimcirq.JSON_RESOLVERS_DICT` if installed or an empty dictionary otherwise.

Reply:
I think it's only for operations with a module name of stimcirq, but, still, this is a good idea, and done.